app/controllers/books_controller.rb

```diff
@@ -0,0 +1,54 @@
+class BooksController < ApplicationController
+
+ def index
+   @books = Book.all.order(:upvotes).reverse
+ end
+
+ def show
+   @book = Book.find(params[:id])
+ end
+
+ def new
+   @book = Book.new
+ end
+
+ def edit
+   @book = Book.find(params[:id])
+ end
+
+ def create
+   @book = Book.new(book_params)
+   if @book.save
+     redirect_to book_path(@book.id)
+   else
+     render :new
+   end
+ end
+
+ def update
+   @book = Book.update(params[:id], book_params)
+   if @book.save
+     redirect_to book_path(@book.id)
+   else
+     render :edit
+   end
+ end
+
+ def destroy
+  Book.find(params[:id]).destroy
+  redirect_to books_path
+ end
+
+ def upvote
+   book = Book.find(params[:id])
```

Comment:
The same `before_action` could be used for finding your book in this controller :smile: